### PR TITLE
Added address search bar to protest map

### PR DESCRIPTION
**Pull request by guyesh**
added component MapSearchAutocomplete.js - search bar for the main map.

### CODE REVIEW

**Comment by uriklar** [comment]
@guyesh This PR looks great! 

A couple of questions:
1. Can we make it so that the search happens once the user selects from the Autocomplete dropdown? (Instead of having to click the target icon on the left)
2. You've got some merge conflicts to resolve
3. and this is a more general one, now that we have that, we don't really need the modal. We can replace the שינוי כתובת 
button with the מציאת הפגנות באזורי.

**Comment by guyesh** [comment]
1. This is how it works, the icon on the left is for gps location. If you select from the dropdown the location changes.
2. I'll fix it.
3. Generally, we don't need the ' שינוי כתובת' button anymore, but we do want to display the modal to mobile users when they open the website. As it is now the modal pops up when the website is load on smartphone.
So I can **remove** the button 
Or **replace** it with another "location by gps" button.
but I think we should anyway show the modal on mobile.

**Comment by guyesh** [comment]
I've replaced the 'שינוי כתובת' button with 'מצא הפגנות באזורי'.
*the modal is still shown when a device without a location in the local storage gets in.

File: src/App.js [hunk 1]
```diff
@@ -199,8 +208,17 @@ function App() {
                     {/* <SiteMessage to="/project-updates/1" style={{ backgroundColor: '#6ab04c' }}>
                       <span style={{ boxShadow: '0 2px 0 0 #fff', fontSize: 19 }}>מה נעשה עכשיו? עדכון פרוייקט #1</span>
                     </SiteMessage> */}
-                    <Button style={{ width: '100%' }} onClick={() => dispatch({ type: 'setModalState', payload: true })}>
+                    {/*<Button style={{ width: '100%' }} onClick={() => dispatch({ type: 'setModalState', payload: true })}>
```
Comment (guytepper) [inline src/App.js]: You can remove this 

```suggestion
```

File: src/components/Map/Map.js [hunk 2]
```diff
@@ -45,15 +46,23 @@ const MarkersList = ({ markers }) => {
 // Initial map value, before the user provide their coordinates.
 const balfur = [31.7749837, 35.219797];
 
-function AppMap({ markers, coordinates, setMapPosition }) {
+function AppMap({ markers, coordinates, setMapPosition, setCoordinates }) {
+  const addressInputRef = useRef(); //search input ref
```
Comment (guytepper) [inline src/components/Map/Map.js]: Please make it neat:
```suggestion
  const addressInputRef = useRef(); // Search Bar ref
```

File: src/components/Map/Map.js [hunk 3]
```diff
@@ -45,15 +46,23 @@ const MarkersList = ({ markers }) => {
...
+
     <MapWrapper
       center={coordinates.length > 0 ? coordinates : balfur}
       onMoveEnd={(t) => {
         setMapPosition([t.target.getCenter().lat, t.target.getCenter().lng]);
       }}
       zoom={14}
     >
+         <SearchPlaceAutoComplete
+          style={{ zIndex: 1, position: 'absolute', top: '30px' }}
```
Comment (guytepper) [inline src/components/Map/Map.js]: Why not use it on the component itself? Seems like some styles are duplicated inside

File: src/components/Map/Map.js [hunk 4]
```diff
@@ -75,7 +84,11 @@ function AppMap({ markers, coordinates, setMapPosition }) {
     </MapWrapper>
   );
 }
-
+const SearchPlaceAutoComplete = styled(MapSearchAutocomplete)`
+  z-index: 10000;
+  position:absolute;
+  top 30;
```
Comment (guytepper) [inline src/components/Map/Map.js]:
```suggestion
  top: 30px;
```

File: src/components/Map/Map.js [hunk 5]
```diff
@@ -75,7 +84,11 @@ function AppMap({ markers, coordinates, setMapPosition }) {
     </MapWrapper>
   );
 }
-
+const SearchPlaceAutoComplete = styled(MapSearchAutocomplete)`
+  z-index: 10000;
+  position:absolute;
```
Comment (guytepper) [inline src/components/Map/Map.js]: Space:
```suggestion
  position: absolute;
```

File: src/components/MapSearchAutocomplete/MapSearchAutocomplete.js [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+        <li style={{ margin: '10px 10px 5px', textAlign: 'left' }}>
+          <img src="/icons/powered_google.png" alt="Powered by Google" style={{ width: 100 }} />
+        </li>
+      </>
+    );
+  };
+
+  return (
+    <Combobox style={{ position: 'absolute', zIndex: 10000, width: '100%' }} onSelect={handleSelect} aria-labelledby="demo">
+      <LocationIcon src="/icons/marker-purple.svg" alt="marker icon" />
```
Comment (guytepper) [inline src/components/MapSearchAutocomplete/MapSearchAutocomplete.js]: No need for `alt` tag here
```suggestion
      <LocationIcon src="/icons/marker-purple.svg" />
```

File: src/components/MapSearchAutocomplete/MapSearchAutocomplete.js [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+      <GpsIcon src="/icons/gps.svg" alt="gps icon" onClick={() => getUserPosition()} />
```
Comment (guytepper) [inline src/components/MapSearchAutocomplete/MapSearchAutocomplete.js]:
```suggestion
      <GpsIcon src="/icons/gps.svg" alt="" onClick={() => getUserPosition()} />
```

File: src/components/MapSearchAutocomplete/MapSearchAutocomplete.js [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+        {suggestions}
+        <li style={{ margin: '10px 10px 5px', textAlign: 'left' }}>
+          <img src="/icons/powered_google.png" alt="Powered by Google" style={{ width: 100 }} />
+        </li>
+      </>
+    );
+  };
+
+  return (
+    <Combobox style={{ position: 'absolute', zIndex: 10000, width: '100%' }} onSelect={handleSelect} aria-labelledby="demo">
```
Comment (guytepper) [inline src/components/MapSearchAutocomplete/MapSearchAutocomplete.js]: What's the demo is for?

File: src/components/MapSearchAutocomplete/MapSearchAutocomplete.js [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+  width: 70%;
+  margin-bottom: 10px;
+  font-size: 16px;
+  padding:10px 50px;
+ 
+  border: 0px;
+  border-radius:50px;
+  -webkit-appearance: none;
+
+  -webkit-box-shadow: 14px 19px 22px -7px rgba(0,0,0,0.19);
```
Comment (guytepper) [inline src/components/MapSearchAutocomplete/MapSearchAutocomplete.js]: No need for prefix.

File: src/components/MapSearchAutocomplete/MapSearchAutocomplete.js (hunk 7 above)
Comment (guytepper) [inline src/components/MapSearchAutocomplete/MapSearchAutocomplete.js]: `getUserPosition` is duplicated 3 places over the source code.  
Can you think of a better, reusable way to use it?

File: src/components/MapSearchAutocomplete/MapSearchAutocomplete.js [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+*/
+  const updateStreetAddress = (address) => setStreetAddress && setStreetAddress(address);
+
+  const handleInput = (e) => {
+    const charactersThreshold = 3;
+    const term = e.target.value;
+    const shouldFetch = term.length >= charactersThreshold;
+    setValue(term, shouldFetch);
+    updateStreetAddress(term);
+  };
```
Comment (guytepper) [inline src/components/MapSearchAutocomplete/MapSearchAutocomplete.js]: Is there a better way to reuse all the copied code from `PlaceAutocomplete`?

File: src/components/MapSearchAutocomplete/MapSearchAutocomplete.js [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+        <ComboboxList>{status === 'OK' && renderSuggestions()}</ComboboxList>
+      </ComboboxPopover>
+    </Combobox>
+  );
+}
+
+const ComboboxInputWrapper = styled(ComboboxInput)`
+  z-index: 10000;
+  position:absolute;
+  top 20px;
```
Comment (guytepper) [inline src/components/MapSearchAutocomplete/MapSearchAutocomplete.js]: Missing `:`

File: src/components/MapSearchAutocomplete/MapSearchAutocomplete.js [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+`;
+const GpsIcon = styled.img`
+  z-index: 10001;
+  top: 29px;
+  left: calc(15% + 17px);
+  position: absolute;
+  height: 20px;
+  user-select: none;
+
+  /*logo is white*/
```
Comment (guytepper) [inline src/components/MapSearchAutocomplete/MapSearchAutocomplete.js]: Which logo? It's not clear. Also please make the comment neater :P

File: src/components/MapSearchAutocomplete/MapSearchAutocomplete.js [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+  -webkit-filter: invert(100%); /* Safari/Chrome */
```
Comment (guytepper) [inline src/components/MapSearchAutocomplete/MapSearchAutocomplete.js]: No need for prefixing. 
```suggestion
```

File: src/components/Modal/Modal.js [hunk 14]
```diff
@@ -43,9 +43,11 @@ function Modal({ isOpen, setIsOpen, coordinates, setCoordinates }) {
     return () => clearTimeout(timeout);
   }, [addressInputDisplay]);
 
+  const [ImageLoaded, setImageLoaded] = useState(false);
```
Comment (guytepper) [inline src/components/Modal/Modal.js]: Remove

File: src/index.css [hunk 15]
```diff
@@ -71,6 +73,7 @@ input {
 
 li {
   margin-bottom: 5px;
+
```
Comment (guytepper) [inline src/index.css]: ?

File: src/components/MapSearchAutocomplete/MapSearchAutocomplete.js [hunk 16]
```diff
@@ -0,0 +1,155 @@
...
+  position: absolute;
+  height: 20px;
+
+  user-select: none;
+
+  @media (min-width: 768px) {
+    right: calc(10% + 20px);
+  }
+`;
+const GpsIcon = styled.img`
```
Comment (guytepper) [inline src/components/MapSearchAutocomplete/MapSearchAutocomplete.js]: Change to GPSIcon
```suggestion
const GPSIcon = styled.img`
```

File: src/components/MapSearchAutocomplete/MapSearchAutocomplete.js [hunk 17]
```diff
@@ -0,0 +1,155 @@
...
+  z-index: 10001;
```
Comment (guytepper) [inline src/components/MapSearchAutocomplete/MapSearchAutocomplete.js]: Can you use more distinct and clear `z-index` values? Like `100`, `200`..

File: src/components/MapSearchAutocomplete/MapSearchAutocomplete.js [hunk 18]
```diff
@@ -0,0 +1,155 @@
...
+  top: 29px;
+  left: calc(15% + 17px);
+  position: absolute;
+  height: 20px;
+  user-select: none;
```
Comment (guytepper) [inline src/components/MapSearchAutocomplete/MapSearchAutocomplete.js]: Display correct pointer.
```suggestion
  user-select: none;
  cursor: pointer;
```

File: src/components/MapSearchAutocomplete/MapSearchAutocomplete.js [hunk 19]
```diff
@@ -0,0 +1,155 @@
...
+const ComboboxInputWrapper = styled(ComboboxInput)`
+  z-index: 10000;
+  position:absolute;
+  top 20px;
+  left:15%;
+  right:15%;
+  width: 70%;
+  margin-bottom: 10px;
+  font-size: 16px;
+  padding:10px 50px;
```
Comment (guytepper) [inline src/components/MapSearchAutocomplete/MapSearchAutocomplete.js]: I'll add `stylelint`, until then please follow https://github.com/necolas/idiomatic-css#declaration-order

**Review by guytepper — CHANGES_REQUESTED** [review]
Hi, great work! Still some work to be done though, mainly with reusing the code. 
I don't want us to maintain code for 2 components that share the same logic.

Also, some notes for the CSS:
- No need to prefix
- I want us to follow [idiomatic-css](https://github.com/necolas/idiomatic-css) style guide. I'll put `stylelint` up for us but until then, give it a quick read.
- Make sure prettier is running successfully. Is your IDE showing you alerts when something is wrong? The missing `:` caused prettier not to run.

 
Screen Shot 2020-10-15 at 8 02 45

Also, please remove the leftovers files & code from the Modal PR :)

File: src/components/MapSearchAutocomplete/MapSearchAutocomplete.js (hunk 17 above)
Comment (guyesh) [reply src/components/MapSearchAutocomplete/MapSearchAutocomplete.js]: the map is using high numbers in z-index I've put 10000 in the input field and the icon should be higher.
I'll add a note to make it clear.

**Review by guytepper — APPROVED** [review]
Sorry for taking so long to review this. 
I've pushed some changes to reflect the latest changes in `master`. 
Thanks! It looks great.

**Pull request merged**